### PR TITLE
Migrate Gradle Plugin out of Bloop

### PR DESCRIPTION
**Pull request by ckipp01**
This pr takes the Gradle integration out of the Bloop codebase and moves it into here.

The functionality of the plugin remains the same, the usage is the same, and the artifact name is the same. We stick with sbt instead of switching to Gradle mainly because there isn't a huge benefit in changing like there was in the [Maven integration](https://github.com/scalacenter/bloop-maven-plugin).

One thing to note is that in the tests we do remove some functionality that is more integration-like that actually extracts the Gradle build definition and then also relies on `frontend` to compile the project. We don't want to lose that, but my plan is to actually move this functionality into more of an "integration suite" utilizing buildpress. So then we'll be able to test both the integrations and make sure they are able to generate bloop definitions and also then compile the projects. I'll create a follow-up issue for this.

Refs: https://github.com/scalacenter/bloop/issues/1680

### CODE REVIEW

File: src/main/scala/bloop/integrations/gradle/BloopPlugin.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import org.gradle.api.Project
+
+/**
+ * Main entry point of the gradle bloop plugin.
+ *
+ * The bloop plugin defines two tasks:
+ *
+ *   1. `configureBloopInstall`: responsible to set up the environment and
+ *      force artifact resolution. 2. `bloopInstall`: responsible of generating
+ *      the bloop config files from the configured data.
```
Comment (adpi2) [inline src/main/scala/bloop/integrations/gradle/BloopPlugin.scala:17]: Something is wrong with the formatting here.

```suggestion
 * 1. `configureBloopInstall`: responsible to set up the environment and
 * force artifact resolution.
 * 2. `bloopInstall`: responsible of generating the bloop config files from
 * the configured data.
```

File: .scalafmt.conf [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+runner.dialect = scala213
+project.git = true
+docstrings.style = Asterisk
+docstrings.wrap = no
+assumeStandardLibraryStripMargin = true
+newlines.beforeCurlyLambdaParams = multilineWithCaseOnly
+align.tokens=[]
+align.openParenCallSite = false
+align.openParenDefnSite = false
+binPack.literalArgumentLists = true
```
Comment (adpi2) [inline .scalafmt.conf:11]: Why not keep the `maxColumn = 100` from scalacenter/bloop.

https://github.com/scalacenter/bloop/blob/f736b2a55a79307f4985ea93408d049466accb88/.scalafmt.conf#L3

**Review by adpi2 — APPROVED** [review]
LGTM

File: src/main/scala/bloop/integrations/gradle/BloopPlugin.scala (hunk 1 above)
Comment (ckipp01) [reply src/main/scala/bloop/integrations/gradle/BloopPlugin.scala:17]: Ah weird, fixed this.

File: .scalafmt.conf (hunk 2 above)
Comment (ckipp01) [reply .scalafmt.conf:11]: Ah, must of just missed it. I'll add it back in.

**Pull request merged**